### PR TITLE
Do not omit type and name while serializing the resource object.

### PR DESCRIPTION
**Pull request by ttlcgl**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by johanste** [comment]
There is no description as to why this change is needed, and as such I'm having a hard time understanding what the broken scenario is. The change looks odd to me as the name and id property of a tracked resource is indeed read-only.

File: specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json [hunk 1]
```diff
@@ -1456,12 +1456,10 @@
         },
         "name": {
           "type": "string",
-          "readOnly": true,
```
[the remark this answers is not in the document]
Comment (RyanBensonMSFT) [reply specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: name, id and type should all be readonly (please add readonly to id as well). These fields should be generated from the URI on resource creation and are not required as input from the user.

[the remark this answers is not in the document]
Comment (RyanBensonMSFT) [reply specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json]: We talked offline, approved

**Review by ryansbenson — APPROVED** [review]
We talked offline, approved

**Comment by johanste** [comment]
Since this may cause breaking changes in generated SDKs, I'd like to better understand the justification. @RyanBensonMSFT, @tatlicioglu, can you please share why this is needed?

**Comment by RyanBensonMSFT** [comment]
Typically performing CRUD operations on nested resources within the context of a parent resource isn't supported across ARM. If you look at MS.Compute/VM/Extensions, they're returned in the Get of a parent VM (as an optimized read) but you can't change them from that context, you have to perform CRUD on each extension directly. This allows for native support of core ARM constructs like RBAC, template deployment, policy... 
After talking to Mehmet, I understand where you are now and understand this mitigation.

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by johanste** [comment]
But this affects more than deep put scenarios, no? It also makes the properties at the parent resource become non-read-only?

**Pull request merged**